### ISSUE TITLE
{ Error: Command failed: gm convert: Option '-crop' requires an argument or argument is malformed.

### ISSUE THREAD

**Issue by desireend** [maxenxe/HQ-Trivia-Bot#4]

**Comment by maxenxe** [comment]
Could you provide me a more detailed error ?

**Comment by maxenxe** [comment]
https://github.com/maxenxe/HQ-Trivia-Bot/issues/5#issuecomment-371536727
Will probably fix your error

**Comment by shabbir1215** [comment]
C:\Users\test\Desktop\HQ-Trivia-Bot-master>node bot.js
Screenshot done
Answers resized
node-tesseract: Running 'tesseract answers.jpg C:\Users\test\AppData\Local\Temp\tmp-6452MMHknBgykeCl'
node-tesseract: Deleting 'C:\Users\test\AppData\Local\Temp\tmp-6452MMHknBgykeCl.txt'
node-tesseract: Preprocessor cleanup
im surren

Question resized
node-tesseract: Running 'tesseract question.jpg C:\Users\test\AppData\Local\Temp\tmp-6452K0rj6nR22UYy'
node-tesseract: Deleting 'C:\Users\test\AppData\Local\Temp\tmp-6452K0rj6nR22UYy.txt'
node-tesseract: Preprocessor cleanup
The question is ""

C:\Users\test\Desktop\HQ-Trivia-Bot-master>

seems to be working but it doesnt seem to quite capture the question??

**Comment by maxenxe** [comment]
I think this is because of the coordinates, did you correctly followed the readme ?

**Comment by shabbir1215** [comment]
Yes i did and after the commands a screenshot saves in the folder showing what it copied and the screenshot seems fine

**Comment by maxenxe** [comment]
Yep but did you correctly modified the coordinates in the code ? How did you got them ?

**Comment by shabbir1215** [comment]
var screenshot = require('desktop-screenshot');
var gm = require('gm');
var nodecr = require('nodecr');
var google = require('google');
var request = require("request");
var cheerio = require('cheerio');
var question;
var answers;
var pointsAnswer0;
var pointsAnswer1;
var pointsAnswer2;

screenshot("screenshot.jpg", {width: 1600}, function(error, complete) {
    if(error)
        console.log("Screenshot failed", error);
    else
        console.log("Screenshot done");
        gm('screenshot.jpg')
          .crop('815', '501', '817', '863')
          .write('answers.jpg', function (err) {
            if (!err) {
              console.log("Answers resized");
              nodecr.process('answers.jpg',function(err, text) {
                if(err) {
                  console.error(err);
                } else {
                  answers = text.replace(/^\s*$[\n\r]{1,}/gm, '');
                  answers = answers.split(/\n/g);
                  answers = answers.filter(function(e){ return e === 0 || e });
                  for(var i = 0; i < answers.length; i++) {
                    console.log(answers[i]);
                  }
                  gm('screenshot.jpg')
                    .crop('725', '300', '101', '545')
                    .write('question.jpg', function (err) {
                      if (!err) {
                        console.log("Question resized");
                        nodecr.process('question.jpg',function(err, text) {
          	              if(err) {
          		              console.error(err);
          	              } else {
                            question = text.replace(/\n/g, " ");
                            console.log('The question is "' + question + '"');
                            google.resultsPerPage = 25;
                            var nextCounter = 0;

                            google(question, function (err, res){
                              if (err) console.error(err)

                              for (var i = 0; i < res.links.length; ++i) {
                                var link = res.links[i];
                                if(link.link != null) {
                                  if(link.link.indexOf("wikipedia") != -1) {
                                    console.log("Wikipedia link found: " + link.link);
                                    request(link.link, function (error, response, body) {
                                      if (!error) {
                                        var $ = cheerio.load(body);
                                        if($('body').text().toLowerCase().indexOf(answers[0].toLowerCase()) != -1) {
                                          console.log(answers[0] + " found in Wikipedia page!")
                                          pointsAnswer0++;
                                        }

                                        if($('body').text().toLowerCase().indexOf(answers[1].toLowerCase()) != -1) {
                                          console.log(answers[1] + " found in Wikipedia page!")
                                          pointsAnswer1++;
                                        }

                                        if($('body').text().toLowerCase().indexOf(answers[2].toLowerCase()) != -1) {
                                          console.log(answers[2] + " found in Wikipedia page!")
                                          pointsAnswer2++;
                                        }
                                      } else {
                                        console.log("Error: " + error);
                                      }
                                    });
                                  }
                                }
                                if(link.title.indexOf(answers[0]) != -1) {
                                  pointsAnswer0++;
                                  console.log(answers[0] + " found!");
                                }
                                if(link.title.indexOf(answers[1]) != -1) {
                                  pointsAnswer1++;
                                  console.log(answers[1] + " found!");
                                }
                                if(link.title.indexOf(answers[2]) != -1) {
                                  pointsAnswer2++;
                                  console.log(answers[2] + " found!");
                                }
                                if(link.description.indexOf(answers[0]) != -1) {
                                  pointsAnswer0++;
                                  console.log(answers[0] + " found!");
                                }
                                if(link.description.indexOf(answers[1]) != -1) {
                                  pointsAnswer1++;
                                  console.log(answers[1] + " found!");
                                }
                                if(link.description.indexOf(answers[2]) != -1) {
                                  pointsAnswer2++;
                                  console.log(answers[2] + " found!");
                                }
                              }

                              if (nextCounter < 4) {
                                nextCounter += 1
                                if (res.next) res.next()
                              }
                            })
          	              }
                        });
                      } else {
                        console.log(err);
                      }
                    });
                }
              });
            } else {
              console.log(err);
            }
          });
});

**Comment by shabbir1215** [comment]
put in all the coordinates as told

**Comment by maxenxe** [comment]
Coordinates are like this: (a,b,c,d), try changing them so they look (c,d,a,b)

**Comment by shabbir1215** [comment]
I tried changing them, it still doesn't work.

**Comment by maxenxe** [comment]
Try increasing and decreasing the values, testing and looking at answers.jpg or question.jpg

**Comment by shabbir1215** [comment]
i have been trying that it does not move but when it does it covers a whole lot of the screen

**Comment by maxenxe** [comment]
It cannot do that because the crop function take as arguments (width,height,x,y) and the x, y are the coordinates from top left point of the cropped image
